src/search/pdbs/utils.cc

```diff
@@ -44,16 +84,21 @@ void dump_pattern_collection_generation_statistics(
     const string &identifier,
     utils::Duration runtime,
     const PatternCollectionInformation &pci,
-    bool dump_collection) {
+    bool dump_collection,
+    int collection_size) {
     const PatternCollection &pattern_collection = *pci.get_patterns();
     if (dump_collection) {
         utils::g_log << identifier << " collection: " << pattern_collection << endl;
     }
     utils::g_log << identifier << " number of patterns: " << pattern_collection.size()
                  << endl;
-    utils::g_log << identifier << " total PDB size: "
-                 << compute_total_pdb_size(pci.get_task_proxy(), pattern_collection)
-                 << endl;
+    utils::g_log << identifier << " total PDB size: ";
+    if (collection_size != -1) {
+        utils::g_log << collection_size << endl;
+    } else {
+        utils::g_log << compute_total_pdb_size(
+            pci.get_task_proxy(), pattern_collection) << endl;
```

Comment:
I'd set `collection_size` to the correct value and then just output that variable.

Reply:
Is there a reason for the two cases in the first place? If the second one cannot always be used, then the reason is not obvious enough. If we're worried about the time it takes to output statistics, then we probably output too many statistics.

Reply:
I just thought that it is unnecessary to re-compute the size of all PDBs of a collection if that value is already known by the generating method. This is why I added an optional size parameter which is printed if given and computed (as before) otherwise. I know it is not really nice and we can change this back if you prefer. (To clarify: this is not about worrying about too much output, but about unnecessary computation time.)

Reply:
I thought that was the rationale, but I'm not sure it's a good idea. I would view this as essentially debug output, and I don't think it would ever be time-critical except in cases where we produce excessive amounts of output. The drawbacks are a more convoluted interface for the function and implementation, the fact that it's fast sometimes but not always, and the fact that the output could be wrong if a wrong number is passed in. None of them critical, but I'd prefer to keep it simple, or alternatively track the total size for all pattern collections in a member variable.